File: repertoire.server/models/album.go
package models

import (
	"time"

	"github.com/google/uuid"
)

type Album struct {
	ID        uuid.UUID  `gorm:"primaryKey; type:uuid; <-:create" json:"id"`
	Title     string     `gorm:"size:100; not null" json:"title"`
	CreatedAt time.Time  `gorm:"default:current_timestamp; not null; <-:create" json:"createdAt"`
	UpdatedAt time.Time  `gorm:"default:current_timestamp; not null" json:"updatedAt"`
	UserID    uuid.UUID  `gorm:"foreignKey:UserID; references:ID; not null" json:"-"`
	ArtistID  *uuid.UUID `json:"-"`
	Artist    Artist     `json:"-"`
	Songs     []Song     `json:"-"`
}

File: repertoire.ui/src/types/models/Album.ts
import Song from './Song.ts'
import Artist from './Artist.ts'

export default interface Album {
  id: string
  title: string
  imageUrl?: string
  releaseDate?: string
  artist?: Artist
  songs: Song[]
  songsCount: number

  rehearsals: number
  confidence: number
  progress: number
  lastTimePlayed?: string

  createdAt: string
  updatedAt: string
}
